src/js/actions/layers.js

```diff
@@ -1296,9 +1296,16 @@ define(function (require, exports) {
                 var allLayerIDs = collection.pluck(results, "ID")
                     .reduce(function (allLayerIDs, layerIDs) {
                         return allLayerIDs.concat(layerIDs);
-                    }, []);
+                    }, []),
+                    allLayers = allLayerIDs.map(function (currentVal) {
+                        return { id: currentVal }
```

Comment:
That is true, sadly the "result" that we have here is not directly compatible to `Layer|Immutable.Iterable.<Layer>` so i made a structure that matches Layer for the purposes of select. If you think it's important i could make a more fully formed layer structure